tests/components/twitch/__init__.py

```diff
@@ -1,246 +1,55 @@
 """Tests for the Twitch component."""
 
-import asyncio
 from collections.abc import AsyncGenerator, AsyncIterator
-from dataclasses import dataclass
-from datetime import datetime
-
-from twitchAPI.object.api import FollowedChannelsResult, TwitchUser
-from twitchAPI.twitch import (
-    InvalidTokenException,
-    MissingScopeException,
-    TwitchAPIException,
-    TwitchAuthorizationException,
-    TwitchResourceNotFound,
-)
-from twitchAPI.type import AuthScope, AuthType
+from typing import Any, TypeVar
 
+from twitchAPI.object.base import TwitchObject
+
+from homeassistant.components.twitch import DOMAIN
 from homeassistant.core import HomeAssistant
 
-from tests.common import MockConfigEntry
+from tests.common import MockConfigEntry, load_json_array_fixture
 
 
 async def setup_integration(hass: HomeAssistant, config_entry: MockConfigEntry) -> None:
     """Fixture for setting up the component."""
     config_entry.add_to_hass(hass)
 
-    await hass.config_entries.async_setup(config_entry.entry_id)
-
-
-def _get_twitch_user(user_id: str = "123") -> TwitchUser:
-    return TwitchUser(
-        id=user_id,
-        display_name="channel123",
-        offline_image_url="logo.png",
-        profile_image_url="logo.png",
-        view_count=42,
-    )
-
-
-async def async_iterator(iterable) -> AsyncIterator:
-    """Return async iterator."""
-    for i in iterable:
-        yield i
-
-
-@dataclass
-class UserSubscriptionMock:
-    """User subscription mock."""
-
-    broadcaster_id: str
-    is_gift: bool
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
 
 
-@dataclass
-class FollowedChannelMock:
-    """Followed channel mock."""
+TwitchType = TypeVar("TwitchType", bound=TwitchObject)
 
-    broadcaster_login: str
-    followed_at: str
 
+class TwitchIterObject:
```

Comment:
```suggestion
class TwitchIterObject(Generic[TwitchType]):
```

Don't miss the generic type here. Otherwise it won't work.